tests/Functional/Action/LtiInstance/CreateActionTest.php

```diff
@@ -0,0 +1,134 @@
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2022 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Tests\Functional\Action\LtiInstance;
+
+use OAT\SimpleRoster\Entity\LtiInstance;
+use OAT\SimpleRoster\Repository\LtiInstanceRepository;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class CreateActionTest extends AbstractLtiInstanceTest
+{
+    protected string $url = '/api/v1/lti-instances';
+    protected string $method = Request::METHOD_POST;
+
+    public function testInvalidAuth(): void
+    {
+        $this->kernelBrowser->request(
+            $this->method,
+            $this->url,
+            [],
+            [],
+            ['PHP_AUTH_USER' => 'wrong', 'PHP_AUTH_PW' => 'invalid'],
+            ''
+        );
+
+        self::assertEquals(
+            Response::HTTP_UNAUTHORIZED,
+            $this->kernelBrowser->getResponse()->getStatusCode()
+        );
+    }
+
+    /**
+     * @dataProvider provideInvalidRequests
+     */
+    public function testValidation(array $body): void
+    {
+        $this->kernelBrowser->request(
+            $this->method,
+            $this->url,
+            [],
+            [],
+            [],
+            json_encode($body)
+        );
+
+        self::assertEquals(
+            Response::HTTP_BAD_REQUEST,
+            $this->kernelBrowser->getResponse()->getStatusCode()
+        );
+    }
+
+    public function testValidRequest(): void
+    {
+        $this->kernelBrowser->request(
+            $this->method,
+            $this->url,
+            [],
+            [],
+            [],
+            json_encode([
+                "label" => "TestLabel",
+                "lti_link" => "http://test.test",
+                "lti_key" => "test1",
+                "lti_secret" => "test2"
+            ])
+        );
+
+        self::assertEquals(
+            Response::HTTP_ACCEPTED,
+            $this->kernelBrowser->getResponse()->getStatusCode()
+        );
+
+        /**@var LtiInstanceRepository */
+        $repository = $this->getRepository(LtiInstance::class);
+
+        self::assertNotNull($repository->findOneBy(['label' => 'TestLabel']));
+
+        self::assertEquals(count($this->data) + 1, count($repository->findAll()));
+    }
+
+    public function provideInvalidRequests(): array
+    {
+        return [
+            'no_lti_secret' => [
+                'body' => [
+                    "label" => "TestLabel",
+                    "lti_link" => "http://test.test",
+                    "lti_key" => "test1",
+                ]
+            ],
+            'no_lti_key' => [
+                'body' => [
+                    "label" => "TestLabel",
+                    "lti_link" => "http://test.test",
+                    "lti_secret" => "test2"
```

Comment:
Please add trailing commas where they missed in this array.

Reply:
fixed